interfaces/acados_template/acados_template/c_templates_tera/matlab_templates/mex_solver.in.m

```diff
@@ -115,6 +115,23 @@ function set_params_sparse(varargin)
             end
         end
 
+
+        function set_p_global(varargin)
+            % usage:
+            % ocp.set_p_global(val)
+            % Sets p_global to val and precomputes all parts of the CasADi graphs of all other functions that only depend on p_global.
+            obj = varargin{1};
+            val = varargin{2};
+
+            if nargin == 2
+                acados_mex_set_p_global_{{ name }}(obj.C_ocp, val);
```

Comment:
Maybe better to move these checks up to `AcadosOcpSolver`. Should make it easier to debug for the user.
At some point (before the overhaul), I thought we just use the templated solver class. But by now I think we want to keep `AcadosOcpSolver`